llama-index-integrations/indices/llama-index-indices-managed-vertexai/llama_index/indices/managed/vertexai/base.py

```diff
@@ -0,0 +1,192 @@
+"""Vertex AI Managed index.
+
+A managed Index - where the index is accessible via some API that
+interfaces a managed service.
+
+"""
+
+import re
+from importlib import metadata
+from typing import Any, Optional, Sequence
+
+from google.cloud.aiplatform_v1beta1 import ImportRagFilesResponse
+from google.cloud.aiplatform import telemetry
+
+import vertexai
+from vertexai.preview import rag
+
+from llama_index.core.base.base_query_engine import BaseQueryEngine
+from llama_index.core.base.base_retriever import BaseRetriever
+from llama_index.core.indices.managed.base import BaseManagedIndex
+from llama_index.core.schema import BaseNode, Document
+
+
+def get_user_agent(module: Optional[str] = None) -> str:
+    r"""Returns a custom user agent header.
+
+    Args:
+        module (Optional[str]):
+            Optional. The module for a custom user agent header.
+
+    Returns:
+        Tuple[str, str]
+    """
+    try:
+        llama_index_version = metadata.version("llama-index")
+    except metadata.PackageNotFoundError:
+        llama_index_version = "0.0.0"
+    client_library_version = (
+        f"{llama_index_version}-{module}" if module else llama_index_version
+    )
+    return f"llama-index/{client_library_version}"
+
+
+class VertexAIIndex(BaseManagedIndex):
+    """Vertex AI Index.
+
+    The Vertex AI RAG index implements a managed index that uses Vertex AI as the backend.
+    Vertex AI performs a lot of the functions in traditional indexes in the backend:
+    - breaks down a document into chunks (nodes)
+    - Creates the embedding for each chunk (node)
+    - Performs the search for the top k most similar nodes to a query
+    - Optionally can perform summarization of the top k nodes
+
+    Args:
+        show_progress (bool): Whether to show tqdm progress bars. Defaults to False.
+    """
+
+    def __init__(
```

Comment:
I guess it might be a little unexpected to some users that neither nodes or document objects are supported in this index. Just a heads up I suppose, but not a huge deal